src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -89,15 +90,19 @@ public static int MainInternal(string[] args, CommandOutputLogger log)
                 tokenSource.CancelAfter(TimeSpan.FromMinutes(DotnetPackageSearchTimeOut));
                 int exitCodeValue = 0;
 
+                CliConfiguration config = new(rootCommand);
+                ParseResult parseResult = rootCommand.Parse(args, config);
+
                 try
                 {
-                    CliConfiguration config = new(rootCommand);
-                    ParseResult parseResult = rootCommand.Parse(args, config);
-                    exitCodeValue = parseResult.InvokeAsync(tokenSource.Token).GetAwaiter().GetResult();
+                    exitCodeValue = parseResult.Invoke();
                 }
                 catch (Exception ex)
                 {
-                    log.LogError(ex.Message);
+                    LogException(ex, log);
+                    var tokenList = parseResult.Tokens.TakeWhile(token => token.Type == CliTokenType.Argument || token.Type == CliTokenType.Command || token.Type == CliTokenType.Directive).Select(t => t.Value).ToList();
+                    tokenList.Add("-h");
+                    rootCommand.Parse(tokenList).Invoke();
                     exitCodeValue = 1;
```

Comment:
>exitCodeValue = 1; [](http://example.com/codeflow?start=20&length=18)

Does returning failure continue to make sense now that we are falling back to known values?

Reply:
We reach here after encountering an error as a result we make sure that is communicated to the caller by setting exitCodeValue = 1. The code above prints out a help output in addition to returning an exit code value of 1.